samples/ck-rx65n/basic-sample/src/main.c

```diff
@@ -0,0 +1,424 @@
+    /* Enter the ThreadX kernel.  */
+    tx_kernel_enter();
+}
+
+/* Define what the initial system looks like.  */
+void    tx_application_define(void *first_unused_memory)
+{
+
+	UINT  status;
+
+	/* Initialize the demo printf implementation. */
+    demo_printf_init();
+
+    /* Initialize the NetX system.  */
+    nx_system_initialize();
+
+    /* Create a packet pool.  */
+    status = nx_packet_pool_create(&pool_0, "NetX Main Packet Pool", SAMPLE_PACKET_SIZE,
+                                   (UCHAR *)sample_pool_stack , sample_pool_stack_size);
+
+    /* Check for pool creation error.  */
+    if (status)
+    {
+        printf("nx_packet_pool_create fail: %u\r\n", status);
+        return;
+    }
+
+    /* Create an IP instance.  */
+    status = nx_ip_create(&ip_0, "NetX IP Instance 0",
+                          SAMPLE_IPV4_ADDRESS, SAMPLE_IPV4_MASK,
+                          &pool_0, nx_driver_rx_fit,
+                          (UCHAR*)sample_ip_stack, sizeof(sample_ip_stack),
+                          SAMPLE_IP_THREAD_PRIORITY);
+
+    /* Check for IP create errors.  */
+    if (status)
+    {
+        printf("nx_ip_create fail: %u\r\n", status);
+        return;
+    }
+
+    /* Enable ARP and supply ARP cache memory for IP Instance 0.  */
+    status = nx_arp_enable(&ip_0, (VOID *)sample_arp_cache_area, sizeof(sample_arp_cache_area));
+
+    /* Check for ARP enable errors.  */
+    if (status)
+    {
+        printf("nx_arp_enable fail: %u\r\n", status);
+        return;
+    }
+
+    /* Enable ICMP traffic.  */
+    status = nx_icmp_enable(&ip_0);
+
+    /* Check for ICMP enable errors.  */
+    if (status)
+    {
+        printf("nx_icmp_enable fail: %u\r\n", status);
+        return;
+    }
+
+    /* Enable TCP traffic.  */
+    status = nx_tcp_enable(&ip_0);
+
+    /* Check for TCP enable errors.  */
+    if (status)
+    {
+        printf("nx_tcp_enable fail: %u\r\n", status);
+        return;
+    }
+
+    /* Enable UDP traffic.  */
+    status = nx_udp_enable(&ip_0);
+
+    /* Check for UDP enable errors.  */
+    if (status)
+    {
+        printf("nx_udp_enable fail: %u\r\n", status);
+        return;
+    }
+
+    /* Initialize TLS.  */
+    nx_secure_tls_initialize();
+
+    /* Create sample helper thread. */
+    status = tx_thread_create(&sample_helper_thread, "Demo Thread",
+                              sample_helper_thread_entry, 0,
+                              sample_helper_thread_stack, SAMPLE_HELPER_STACK_SIZE,
+                              SAMPLE_HELPER_THREAD_PRIORITY, SAMPLE_HELPER_THREAD_PRIORITY,
+                              TX_NO_TIME_SLICE, TX_AUTO_START);
+
+    /* Check status.  */
+    if (status)
+    {
+        printf("Demo helper thread creation fail: %u\r\n", status);
+        return;
+    }
+}
+
+/* Define sample helper thread entry.  */
+void sample_helper_thread_entry(ULONG parameter)
+{
+UINT    status;
+ULONG   ip_address = 0;
+ULONG   network_mask = 0;

```

Comment:
shadows a global

Reply:
this code:
```
#ifndef SAMPLE_DHCP_DISABLE
static NX_DHCP          dhcp_0;
NX_DHCP           dhcp_client;
UCHAR             ip_address[4];
UCHAR             network_mask[4];
#endif /* SAMPLE_DHCP_DISABLE  */
```
FYI, I'm curious where it came from. I searched for this UCHAR declaration and I cannot find it in any of the mains from Microsoft's package for that board.